test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua

```diff
@@ -0,0 +1,31 @@
+local netbox = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_rel_paths_in_diag = function(_)
+    local _, err = pcall(function() netbox.self:call('') end)
+    t.assert_not(err.trace[1].file:startswith('/'))
+end
+
+g.test_rel_paths_in_debug_info = function(_)
+    t.skip_if(bit.rshift(bit.band(os.execute('readelf --version'), 0xFF00), 8) == 127)
+
+    local ph = io.popen('readelf --debug-dump=info ' .. arg[-1])
+    for line in ph:lines() do
+        local i = 1
+        for word in line:gmatch('%S+') do
+            if i ~= 2 and i ~= 9 then
+                goto continue
+            end
+            if i == 2 and word ~= "DW_AT_comp_dir" then
+                break
+            end
+            if i == 9 then
+                t.assert_str_matches(word, '^[^/]')
```

Comment:
This script passes with `DEV_BUILD=ON` - looks like this code never runs. Why not simplify path extraction? Something like this, maybe?

```lua
s:gsub('.*DW_AT_comp_dir.*:%s*(.*)$', '%1')
```

Reply:
> This script passes with DEV_BUILD=ON - looks like this code never runs. 

<img width="1233" alt="Screen Shot 2022-10-28 at 19 19 21" src="https://user-images.githubusercontent.com/32464072/198685128-4be644f1-2d50-4402-babb-a55a9f0838ac.png">

It doesn't pass on my machine — looks like you didn't reset your CMake cache.

Reply:
> Why not simplify path extraction? Something like this, maybe? 

Thanks for the suggestion, I couldn't think of an appropriate regular expression.

But initially I aimed for the best performance: with your approach the test now takes 112 seconds vs. 14 seconds of my approach.

Reply:
> It doesn't pass on my machine — looks like you didn't reset your CMake cache.

Now, *that you updated the script*, it doesn't pass on my machine, either. Anyhow, it *should pass* even if `DEV_BUILD` is enabled - the test should just be skipped in this case. I think you should be able to check the build options with `require('tarantool').build.options`.

> But initially I aimed for the best performance: with your approach the test now takes 112 seconds vs. 14 seconds with my approach.

Performance is a good point. If the test takes that long with regular expressions, then we need to do something about the check. Can we speed up the test like this?

```diff
diff --git a/test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua b/test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua
index a03b870665ac..520f8c88b743 100644
--- a/test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua
+++ b/test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua
@@ -13,8 +13,8 @@ g.test_rel_paths_in_debug_info = function(_)
 
     local ph = io.popen('readelf --debug-dump=info ' .. arg[-1])
     for line in ph:lines() do
-        local _, _, path = line:find('.*DW_AT_comp_dir.*:%s*(.*)$')
-        if path ~= nil then
+        if line:find('DW_AT_comp_dir') then
+            local _, _, path = line:find('.*DW_AT_comp_dir.*:%s*(.*)$')
             t.assert_str_matches(path, '^[^/].*')
         end
     end
```

Reply:
Applied your diff, thanks for the suggestion!